packages/minimongo/cursor.js

```diff
@@ -522,7 +522,7 @@ ASYNC_CURSOR_METHODS.forEach(method => {
   Cursor.prototype[asyncName] = function(...args) {
     try {
       this[method].isCalledFromAsync = true;
-      return Promise.resolve(this[method].apply(this, args));
+      return Promise.asyncApply(async () => this[method].apply(this, args));
```

Comment:
The problem I see with this solution is that it relies on Fibers (`asyncApply`), and we're removing it. 

I did some tests with our branch `release-3.0`, and it works as expected:

![image](https://user-images.githubusercontent.com/29639852/236014823-0fcefa48-e980-464b-929e-62f4bee5fdb4.png)

I think that we're getting this result right now because the current release (2.12) still relies on Fibers.

Let's proceed with your changes. I'll just ask you to add a TODO above the return, so we remember to keep the Promise.resolve for the release 3.0. It can be something like:

```
TODO [fibers]: revert this change to "Promise.resolve(this[method].apply(this, args));" in version 3.0
```